docs/spec/relayer-v2/Chain.tla

```diff
@@ -0,0 +1,146 @@
+    \* update clients
+    LET clientUpdatedChain == HandleUpdateClient(chainID, clientCreatedChain, datagrams) IN
+
+    clientUpdatedChain
+    
+(***************************************************************************
+ Connection update operators
+ ***************************************************************************)
+\* Update the connections on chain with chainID, 
+\* using the connection datagrams generated by the relayer
+\* (Handler operators defined in ConnectionHandlers.tla)
+ConnectionUpdate(chainID, chain, datagrams) ==
+    \* update chain with "ConnOpenInit" datagrams
+    LET connOpenInitChain == HandleConnOpenInit(chainID, chain, datagrams) IN
+    \* update chain with "ConnOpenTry" datagrams
+    LET connOpenTryChain == HandleConnOpenTry(chainID, connOpenInitChain, datagrams) IN
+    \* update chain with "ConnOpenAck" datagrams
+    LET connOpenAckChain == HandleConnOpenAck(chainID, connOpenTryChain, datagrams) IN
+    \* update chain with "ConnOpenConfirm" datagrams
+    LET connOpenConfirmChain == HandleConnOpenConfirm(chainID, connOpenAckChain, datagrams) IN
+    
+    connOpenConfirmChain
+
+(***************************************************************************
+ Channel update operators
+ ***************************************************************************)
+\* Update the channels on chain with chainID, 
+\* using the channel datagrams generated by the relayer
+\* (Handler operators defined in ChannelHandlers.tla)
+ChannelUpdate(chainID, chain, datagrams) ==
+    \* update chain with "ChanOpenInit" datagrams
+    LET chanOpenInitChain == HandleChanOpenInit(chainID, chain, datagrams) IN
+    \* update chain with "ChanOpenTry" datagrams
+    LET chanOpenTryChain == HandleChanOpenTry(chainID, chanOpenInitChain, datagrams) IN
+    \* update chain with "ChanOpenAck" datagrams
+    LET chanOpenAckChain == HandleChanOpenAck(chainID, chanOpenTryChain, datagrams) IN
+    \* update chain with "ChanOpenConfirm" datagrams
+    LET chanOpenConfirmChain == HandleChanOpenConfirm(chainID, chanOpenAckChain, datagrams) IN
+    
+    chanOpenConfirmChain
+
+(***************************************************************************
+ Chain update operators
+ ***************************************************************************)
+\* Update chainID with the received datagrams
+\* Supports ICS2 (Clients), ICS3 (Connections), and ICS4 (Channels).
+UpdateChainStore(chainID, datagrams) == 
+    \* ICS 002: Client updates
+    LET lightClientsUpdatedStore == LightClientUpdate(chainID, chainStore, datagrams) IN 
+    \* ICS 003: Connection updates
+    LET connectionsUpdatedStore == ConnectionUpdate(chainID, lightClientsUpdatedStore, datagrams) IN
+    \* ICS 004: Channel updates
+    LET channelsUpdatedStore == ChannelUpdate(chainID, connectionsUpdatedStore, datagrams) IN
+    
+    \* update height
+    LET updatedChainStore == 
+        IF /\ chainStore /= channelsUpdatedStore
```

Comment:
In principle, if incomingDatagrams is not empty, we should increase the height. This would model the fact that IBC datagrams are processed on chain, although they don't necessarily need to lead to state change. On the other side, that would potentially significantly increase the state space, so increasing height only when we effectively make state change by processing datagrams is more meaningful. Maybe we should just add comment explaining this design decision.

Reply:
I agree, increasing the height will imply that we need to assign higher value to `MaxHeight`, which in turn increases the state space. I'll add a comment.